packages/tools/src/tools/displayTools/Contour/contourDisplay.ts

```diff
@@ -180,42 +196,69 @@ async function render(
   );
 }
 
-// function _checkContourNormalsMatchViewport(
-//   annotationUIDsMap: Map<number, Set<string>>,
-//   viewportNormal: Types.Point3
-// ): boolean {
-//   const annotationUIDs = Array.from(annotationUIDsMap.values())
-//     .flat()
-//     .map((uidSet) => Array.from(uidSet))
-//     .flat();
-
-//   // Take up to 3 random annotations to check
-//   const sampleSize = Math.min(3, annotationUIDs.length);
-//   const randomAnnotationUIDs = [];
-//   for (let i = 0; i < sampleSize; i++) {
-//     const randomIndex = Math.floor(Math.random() * annotationUIDs.length);
-//     randomAnnotationUIDs.push(annotationUIDs[randomIndex]);
-//   }
-
-//   for (const annotationUID of randomAnnotationUIDs) {
-//     const annotation = getAnnotation(annotationUID);
-//     if (annotation?.metadata?.viewPlaneNormal) {
-//       const annotationNormal = annotation.metadata.viewPlaneNormal;
-//       // Check if normals are parallel or anti-parallel (dot product close to 1 or -1)
-//       const dotProduct = Math.abs(
-//         viewportNormal[0] * annotationNormal[0] +
-//           viewportNormal[1] * annotationNormal[1] +
-//           viewportNormal[2] * annotationNormal[2]
-//       );
-
-//       if (Math.abs(dotProduct - 1) > 0.01) {
-//         return false;
-//       }
-//     }
-//   }
-
-//   return true;
-// }
+function _checkContourGeometryMatchViewport(
+  geometryIds: string[],
+  viewportNormal: Types.Point3
+): boolean {
+  const geometry = cache.getGeometry(geometryIds[0]);
+
+  if (!geometry) {
+    return false;
+  }
+
+  const geometryData = geometry.data as Types.IContourSet;
+
+  const contours = geometryData.contours;
+  const points = contours[0].points;
+  const point1 = points[0];
+  const point2 = points[1];
+  const point3 = points[2];
+
+  // get the normal from two points
+  let normal = vec3.cross(
+    vec3.create(),
+    vec3.sub(vec3.create(), point2, point1),
+    vec3.sub(vec3.create(), point3, point1)
+  );
+  normal = vec3.normalize(vec3.create(), normal);
+  const dotProduct = vec3.dot(normal, viewportNormal);
+  return Math.abs(dotProduct) > 0.9;
+}
+
+function _checkContourNormalsMatchViewport(
+  annotationUIDsMap: Map<number, Set<string>>,
+  viewportNormal: Types.Point3
+): boolean {
+  const annotationUIDs = Array.from(annotationUIDsMap.values())
+    .flat()
+    .map((uidSet) => Array.from(uidSet))
+    .flat();
+
+  // Use getRandomSampleFromArray to get up to 3 random annotations
+  const randomAnnotationUIDs = utilities.getRandomSampleFromArray(
+    annotationUIDs,
+    3
+  );
+
+  for (const annotationUID of randomAnnotationUIDs) {
+    const annotation = getAnnotation(annotationUID);
+    if (annotation?.metadata?.viewPlaneNormal) {
+      const annotationNormal = annotation.metadata.viewPlaneNormal;
+      // Check if normals are parallel or anti-parallel (dot product close to 1 or -1)
+      const dotProduct = Math.abs(
```

Comment:
Check if viewPlaneNormal is defined - this shouldn't throw if it isn't.  In specific, if you load an SR containing an annotation it shouldn't throw, and I think that case has an empty viewPlaneNorma.   Maybe construct one if it isn't present.